libs/base/src/random/RandomGenerator.cpp

```diff
@@ -27,76 +27,25 @@ CRandomGenerator mrpt::random::randomGenerator;
 // Initialize the generator from a seed
 void CRandomGenerator::MT19937_initializeGenerator(const uint32_t &seed)
 {
-	m_MT19937_data.seed_initialized = true;
-	m_MT19937_data.MT[0] = seed;
-	for (uint32_t i=1;i<624;i++)
-		m_MT19937_data.MT[i] = static_cast<uint32_t>( 1812433253 * (m_MT19937_data.MT[i-1] ^ ( m_MT19937_data.MT[i-1] >> 30 )) + i); // 0x6c078965
+	m_MT19937.seed(seed);
 }
 
-#if defined(_MSC_VER)
-	#pragma warning(push)
-	#pragma warning(disable:4146)
-#endif
-
-// Code from the implementation by Rick Wagner
-//  http://www-personal.umich.edu/~wagnerr/MersenneTwister.html
-inline uint32_t hiBit( const uint32_t u ) { return u & 0x80000000UL; }
-inline uint32_t loBit( const uint32_t u ) { return u & 0x00000001UL; }
-inline uint32_t loBits( const uint32_t u ) { return u & 0x7fffffffUL; }
-inline uint32_t mixBits( const uint32_t u, const uint32_t v ) { return hiBit(u) | loBits(v); }
-inline uint32_t twist( const uint32_t m, const uint32_t s0, const uint32_t s1 ) { return m ^ (mixBits(s0,s1)>>1) ^ (-loBit(s1) & 0x9908b0dfUL); }
-
-#if defined(_MSC_VER)
-	#pragma warning(pop)
-#endif
-
-
 // Generate an array of 624 untempered numbers
 void CRandomGenerator::MT19937_generateNumbers()
 {
```

Comment:
Great!!! I was thinking this was a bit useless with the changes and I should have looked at this closer.